.github/workflows/framework-tests.yaml

```diff
@@ -43,7 +43,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python-version: ["3.8", "3.10", "3.12"]
+        python-version: ["3.8", "3.10", "3.12", "3.13"]
```

Comment:
I'm open to having a new discussion if you want, but on the [last PR](https://github.com/canonical/operator/pull/1315) that updated this we decided to run the tests just on the LTS versions (3.8, 3.10, 3.12), but do the `pip install` on 3.13 (which it already is, but we could remove the pre-release).

I think we should keep "test" and "test-real-pebble" on just 3.8/3.10/3.12, but update "lint" to 3.13 (it's currently still on 3.11).

That said, it might be a few days premature -- looks like the GitHub Action can't find Python 3.13 yet, hence the failed runs.